diag/diag.proto

```diff
@@ -0,0 +1,238 @@
+//
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// This file defines the gNOI APIs used to perform diagnostic operations on a
+// switching box.
+syntax = "proto3";
+
+package gnoi.diag;
+
+// The Diag service exports to main set of RPCs:
+// 1- BERT related RPCs: Used to perform Bit Error Rate Test (BERT)
+//    operations on a set of ports.
+// 2- BURNING related RPCs: Used to perform a vendor-provided Burnin test on the
+//    switch to ensure the box is ready to start serving traffic. Burnin tests
+//    are typically run in the datacenter, as part of turnup or repair workflow.
+// Note: The RPCs defined here are are stateless operations and them
+// failing/passing should not leave any permanent artifact on the switch (unless
+// there is something wrong HW-wise).
+service Diag {
+  // Starts BERT operation on a set of ports. Each BERT operation is uniquely
```

Comment:
When we say `port` here, do we mean `interface` in OpenConfig parlance? Or do we mean a single physical port? I think it'd be good to clarify since it's not necessarily clear whether the `gnoi.Path` messages should correspond to `/interfaces/interface` entries or `/components/component` entries of type `PORT`.

Reply:
Ports refer to a channelized port which we previously identified using a (slot, port, channel). In YANG, I think an interface is a good equivalent. I clarified this in this text.